warehouse/age-off/src/main/java/datawave/ingest/util/cache/watch/FileRuleCacheValue.java

```diff
@@ -0,0 +1,130 @@
+    }
+
+    /**
+     * Gets the file path of this instance.
+     *
+     * @return path for the instance
+     */
+    public Path getFilePath() {
+        return filePath;
+    }
+
+    /**
+     * Check if the cached representation has changes. Changes are determined by checking the baseline modification time when the cached representation was
+     * discovered against the current modification time of the file.
+     *
+     * @return true if there are changes, otherwise false
+     */
+    public boolean hasChanges() {
+        if (ruleRef == null) {
+            return true;
+        }
+        long currentTime;
+        try {
+            currentTime = fs.getFileStatus(filePath).getModificationTime();
+        } catch (IOException e) {
+            log.debug("Error getting file status for: " + filePath, e);
+            return true;
+        }
+        long previousTime = ruleRef.getTimestamp();
+        boolean changed = (currentTime - previousTime) > configuredDiff;
+        if (log.isTraceEnabled()) {
+            log.trace("Changes result: " + changed + ", current time: " + currentTime);
+        }
+        return changed;
+    }
+
+    /**
+     * Creates a new ruleset view of the file. The initial call to the method will lazily create the base rules and return a view of the baseline rules. The
+     * next calls will create new view copies derived from the baseline rules.
+     *
+     * @param scanStart
+     *            the start of a scan operation to use for the ruleset
+     * @param iterEnv
+     *            the iterator environment for the scan
+     * @return a deep copy of the cached {@link AppliedRule} baseline rules
+     * @throws IOException
+     *             if there are errors during the cache value creation, on initial call
+     */
+    public Collection<AppliedRule> newRulesetView(long scanStart, IteratorEnvironment iterEnv) throws IOException {
+        // rule initialization/copies are performed on the calling thread
+        // the base iterator rules will use an iterator environment from the caller (and keep in the AppliedRule)
+        // the deep copy always creates new views of the rules with the caller's iterator environment
+        if (ruleRef == null) {
+            long ts = fs.getFileStatus(filePath).getModificationTime();
+            Collection<FilterRule> rulesBase = loadFilterRules(iterEnv);
+            ruleRef = new FileRuleReference(ts, rulesBase);
+        }
+        return ruleRef.deepCopy(scanStart, iterEnv);
+    }
+
+    @VisibleForTesting
+    Collection<FilterRule> loadFilterRules(IteratorEnvironment iterEnv) throws IOException {
+        AgeOffRuleLoader ruleLoader = new AgeOffRuleLoader(new FileLoaderDependencyProvider(fs, filePath, iterEnv));
+        Collection<FilterRule> rulesBase;
+        try (InputStream in = fs.open(filePath)) {
+            rulesBase = ruleLoader.load(in);
+        } catch (Exception e) {
```

Comment:
update to be a more specific catch

Reply:
Removed general exception catch - `IOException` remained on the method as a checked exception.